### PR TITLE
Add kinesis producer/consumer recipe

### PR DESCRIPTION
**Pull request by sabbyanandan**
The PR includes the simplest demo possible. It should be quick to repeat (not many moving parts).

### CODE REVIEW

File: content/documentation/pages/6-recipes/5-kinesis/1-simple-producer-consumer.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+In this walk-through, though, we will review a simple use-case to showcase how the Kinesis binder can be used with Spring Cloud Stream.
+
+## Prerequisite
+
+The only requirement for the demonstration is the "Access Key", "Secret Key", and "Region" credentials, which can be gathered from your AWS account.
+
+## Applications
+
+As for the sample producer and consumer applications, you can clone the repository at [sabbyanandan/kinesisdemo](https://github.com/sabbyanandan/kinesisdemo) to follow along with the sample walk-through.
```
Comment (artembilan) [inline content/documentation/pages/6-recipes/5-kinesis/1-simple-producer-consumer.md]: Is it really acceptable to link in the official docs your own GitHub repo?
Can't that demo go somewhere in the common one, e.g. https://github.com/spring-cloud/spring-cloud-stream-samples ?

File: content/documentation/pages/6-recipes/5-kinesis/1-simple-producer-consumer.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+<!--KinesisProducer Configuration-->
+
+```yaml
+spring:
+  cloud:
+    stream:
+      bindings:
+        output:
+          destination: test-kinesis-stream
+          group: test-kinesis-stream-group
```
Comment (artembilan) [inline content/documentation/pages/6-recipes/5-kinesis/1-simple-producer-consumer.md]: I think `group` is redundant on the *producer* side.

File: content/documentation/pages/6-recipes/5-kinesis/1-simple-producer-consumer.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+# Amazon Kinesis Sample
+
+The Spring team currently maintains the Spring Cloud Stream Kinesis binder with the help of community contributions.
+You can read more about the binder implementation at [spring-cloud/spring-cloud-stream-binder-aws-kinesis](https://github.com/spring-cloud/spring-cloud-stream-binder-aws-kinesis).
+
+In this walk-through, though, we will review a simple use-case to showcase how the Kinesis binder can be used with Spring Cloud Stream.
+
+## Prerequisite
+
+The only requirement for the demonstration is the "Access Key", "Secret Key", and "Region" credentials, which can be gathered from your AWS account.
```
Comment (artembilan) [inline content/documentation/pages/6-recipes/5-kinesis/1-simple-producer-consumer.md]: I think we also need to say that `spring-cloud-stream-binder-kinesis` dependency must be present.

File: content/documentation/pages/6-recipes/5-kinesis/1-simple-producer-consumer.md (hunk 1 above)
Comment (sabbyanandan) [reply content/documentation/pages/6-recipes/5-kinesis/1-simple-producer-consumer.md]: I will have to figure out how to do that thing.

**Comment by sabbyanandan** [comment]
@artembilan: Thank you for the review! 

> Nothing critical, but I wonder what is the point of such a plain sample...

From what I learned from Labs/pivots, they simply don't know how to use it. My attempt here was to hopefully bootstrap that experience, so folks like them can discover other things on their own.

> Isn't there something what really makes a Kinesis binder much valuable over Kafka or Rabbit?

I don't want to write comparison documentation on what binder can be used and where. That can be another type of a recipe altogether. The scope of this PR is simply to highlight how an SCSt app can be built with this binder, which we do not have it anywhere.

> How about to mention how it works if the app is ran in the AWS EC2 env: https://aws.amazon.com/ec2/?

I have not used it, so not sure I can speak to it. Please feel free to amend the PR with what you think could be useful.

File: content/documentation/pages/6-recipes/5-kinesis/1-simple-producer-consumer.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
```
Comment (artembilan) [inline content/documentation/pages/6-recipes/5-kinesis/1-simple-producer-consumer.md]: BTW, when we run our app in AWS EC2, we don't need credential properties, as well as `region`.
There are determined automaticaly from the AWS environment. 
Might be worth to mention

**Comment by artembilan** [comment]
> they simply don't know how to use it

Let's ask then someone from there to review this as well!
That would be great feedback for us.
The problem is that we really may miss something since it feels for us too obvious.

**Comment by sabbyanandan** [comment]
> Let's ask then someone from there to review this as well!

I originally created https://github.com/sabbyanandan/kinesisdemo to help them with that a few months ago. They already used it as a baseline back when they were doing POCs.

File: content/documentation/pages/6-recipes/5-kinesis/1-simple-producer-consumer.md (hunk 1 above)
Comment (artembilan) [reply content/documentation/pages/6-recipes/5-kinesis/1-simple-producer-consumer.md]: All good, but this.
Would be great to move it into our common sample. Isn't it?

Comment (sabbyanandan) [reply content/documentation/pages/6-recipes/5-kinesis/1-simple-producer-consumer.md]: Yep! Agreed. It bugs me also. Let me figure it out when I get to the office.

Comment (sabbyanandan) [reply content/documentation/pages/6-recipes/5-kinesis/1-simple-producer-consumer.md]: This is now attempted to be ported over to Microsite's samples repo at: https://github.com/spring-cloud/spring-cloud-dataflow-samples/pull/106

**Pull request merged**